src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -561,6 +562,25 @@ class Peer(
                     // no additional inputs or outputs, the splice is only meant to bump fees
                     spliceIn = null,
                     spliceOut = null,
+                    requestRemoteFunding = null,
+                    feerate = feerate
+                )
+                send(WrappedChannelCommand(channel.channelId, spliceCommand))
+                spliceCommand.replyTo.await()
+            }
+    }
+
+    suspend fun purchaseInboundLiquidity(amount: Satoshi, feerate: FeeratePerKw, maxFee: Satoshi, leaseDuration: Int): ChannelCommand.Commitment.Splice.Response? {
```

Comment:
For ease of integration:

```suggestion
    suspend fun purchaseInboundLiquidity(amount: Satoshi, feerate: FeeratePerKw, leaseRates: LeaseRates): ChannelCommand.Commitment.Splice.Response? {
```

Assuming that lease rates are an attribute of the `Peer`, or of `NodeParams`, this could be simplified to:

```suggestion
    suspend fun purchaseInboundLiquidity(amount: Satoshi, feerate: FeeratePerKw): ChannelCommand.Commitment.Splice.Response? {
```